video/decode/vd_lavc.c

```diff
@@ -952,6 +958,40 @@ static enum AVPixelFormat get_format_hwdec(struct AVCodecContext *avctx,
     return select;
 }
 
+static int get_buffer2_direct(AVCodecContext *avctx, AVFrame *pic, int flags)
+{
+    struct dec_video *vd = avctx->opaque;
+    struct vo_buffer_data vbufd = { 0 };
+
+    struct mp_image img;
+    mp_image_copy_fields_from_av_frame(&img, pic);
+    vbufd.par            = &img.params;
+    vbufd.aligned_width  = img.params.w;
+    vbufd.aligned_height = img.params.h;
+    avcodec_align_dimensions2(avctx, &vbufd.aligned_width, &vbufd.aligned_height,
+                              vbufd.stride_alignment);
```

Comment:
Its absolutely needed. Where else would libavcodec handle it? Another avctx field?